service/controller/v18/resource/ipam/create.go

```diff
@@ -104,13 +122,29 @@ func (r *Resource) allocateSubnet(ctx context.Context) (string, error) {
 
 		subnet, err = ipam.Free(r.networkRange, r.allocatedSubnetMask, reservedSubnets)
 		if err != nil {
-			return "", microerror.Maskf(err, "networkRange: %s, allocatedSubnetMask: %s, reservedSubnets: %#v", r.networkRange.String(), r.allocatedSubnetMask.String(), reservedSubnets)
+			return net.IPNet{}, microerror.Maskf(err, "networkRange: %s, allocatedSubnetMask: %s, reservedSubnets: %#v", r.networkRange.String(), r.allocatedSubnetMask.String(), reservedSubnets)
 		}
 
 		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found free subnet %#q", subnet.String()))
 	}
 
-	return subnet.String(), nil
+	return subnet, nil
+}
+
+func (r *Resource) selectRandomAZs(n int) ([]string, error) {
+	if n > len(r.availabilityZones) {
+		return nil, microerror.Maskf(invalidParameterError, "requested nubmer of AZs %d is bigger than number of available AZs %d", n, len(r.availabilityZones))
+	}
+
+	// availabilityZones must be copied so that original slice doesn't get shuffled.
+	shuffledAZs := make([]string, len(r.availabilityZones))
+	copy(shuffledAZs, r.availabilityZones)
+	rand.Shuffle(len(shuffledAZs), func(i, j int) {
```

Comment:
No it isn't, but:
> If Seed is not called, the generator behaves as if seeded by Seed(1).

The `math/rand` is deterministic anyway and when we are only randomizing availability zones (which means we might be selecting commonly 1/3 or 2/3), does it really matter?

I'm totally fine adding this, but I'm wondering if there's any real reason why it would be needed?

Reply:
One reason is e2e tests will always have the same AZs. And in general I think it's a good idea to do the `Seed` call.

Reply:
Ack. On it's way..